### ISSUE TITLE
WaffleAuthenticatorBase not flexible to configure WindowsAuthProviderImpl.continueContextTimeout

### ISSUE THREAD

**Issue by alanlavintman** [Waffle/waffle#194]
I wanted to configure the timeout used by the WindowsAuthProviderImpl instance through the NegotiateAuthenticator. Ideally i would like to define the NegotiateAuthenticator instance in the context.xml as follows: 

```
<Valve className="waffle.apache.NegotiateAuthenticator" principalFormat="sid" roleFormat="sid" allowGuestLogin="false" continueContextTimeout=15000 />
```

The attribute continueContextTimeout is not a field in WaffleAuthenticatorBase, how can i have this flexibility?

**Comment by dblock** [comment]
Generally I think it's not necessary because Windows has built-in timeouts for various phases of negotiation. I believe these tickets expire fast.

That said, expising an expiration time as you describe here is a fine feature request, please feel free to contribute.

**Comment by alanlavintman** [comment]
Thanks a lot for the fast answer, i will make a pull request and evaluate
it on your side :)
Im trying to build the JNA project but there are two unit tests failling on
tomcat8, its about the loginInfo taking it from the context vs receiving it
by parameter (In previous versions for tomcat6 and 7). Am i correct? If so,
can i submit a patch for it as well?

Bests,
Alan.


2015-01-27 17:33 GMT+02:00 Daniel Doubrovkine (dB.) @dblockdotorg <
notifications@github.com>:

**Comment by hazendaz** [comment]
Strange.  The builds on Travis ci are running unit tests just fine.  Can you post the condition you are seeing?

--- Original Message ---

**Comment by hazendaz** [comment]
I forgot Travis ci doesn't run the tests.  So this failure in tests makes sense.

--- Original Message ---

**Comment by alanlavintman** [comment]
Overall all the tests pass just fine !
I did a pull request with a little fix, let me know if it helps. When the
tests pass let me know and i will add the configuration to thre
continuecontxttimeout property :)

**Comment by hazendaz** [comment]
@alanlavintman Can you take a look at your original pull request?  We had some outstanding issues that needed addressed before we pull this in.

**Comment by hazendaz** [comment]
Looking at this again.  Due to non response, I'm going to take this up and add it both to the 1.8.4 release and 1.9.0 upcoming releases.  We had 30 seconds hard-coded internally so this absolutely makes sense to allow from the outside.  What we are missing is documentation and the tests fail at the moment but I've got it rebased on 1.8.x line and I'm trying to work through that.  As soon as I get that done, I'll snag the setup from here for configuration needs, add the necessary changelog for original contributor, and get that applied.

**Comment by hazendaz** [comment]
Seems we haven't actually defined any docs or setups using anything other than the most basic defaults so for this effort I'm not going to spend time there.  I got this applied to 1.8.4-SNAPSHOT and got a change log.  I'll port to the master shortly as well and get these tickets closed.  I'll look more at further documentation needs against master as we are lacking in some areas based on what I see.

**Comment by hazendaz** [comment]
Closing as fixed per #579
